python/ray/train/v2/_internal/execution/failure_handling/default.py

```diff
@@ -14,33 +15,54 @@ class DefaultFailurePolicy(FailurePolicy):
     def __init__(self, failure_config: FailureConfig):
         super().__init__(failure_config)
         self._total_failures = 0
+        self._resize_failure_count = 0
 
     def make_decision(
-        self, worker_group_status: WorkerGroupPollStatus
+        self, worker_group_status: PolicyHandledStatus
     ) -> FailureDecision:
         if not worker_group_status.errors:
             return FailureDecision.NOOP
 
-        self._total_failures += 1
-
-        if self.failure_config.max_failures == -1:
+        if isinstance(worker_group_status, WorkerGroupResizeStatus):
+            self._resize_failure_count += 1
+            if (
+                hasattr(self.failure_config, "resize_failure_limit")
+                and self.failure_config.resize_failure_limit != -1
+                and self._resize_failure_count
+                > self.failure_config.resize_failure_limit
```

Comment:
We do not need to handle this backwards compatibility case since users will import from `ray.train.FailureConfig`, which will be routed to the correct implementation when the V2 env var is set.